### PR TITLE
Feature/112 credential library

### LINKED ISSUE #112: Create a credential helper library
Take the `datastore` [credential helper](https://github.com/verida/datastore/blob/master/src/helpers/credentials.js) and upgrade it to use the new `verida-ts` library.

### PR DESCRIPTION
**Pull request by cmcWebCode40**
- Resolved designed issue as suggested https://github.com/verida/verida-js/issues/113#issuecomment-989507537
- Modified tests to work on the new design

Resolves #112

### CODE REVIEW

File: packages/verifiable-credentials/src/sharing-credential.ts [hunk 1]
```diff
@@ -20,45 +16,20 @@ interface VCResult {
...
-	async issueEncryptedCredential(cred: any): Promise<any> {
-		// Issue a new public, encrypted verida credential
-		const account = await this.context.getAccount();
+	/**
+	 * Method to encrypt and issue credential
+	 * @param didJwtVc
+	 * @returns
+	 */
+
+	async issueEncryptedCredential(didJwtVc: string): Promise<any> {
```
Comment (tahpot) [inline packages/verifiable-credentials/src/sharing-credential.ts:29]: This needs to pass the full `item`.

File: packages/verifiable-credentials/src/sharing-credential.ts [hunk 2]
```diff
@@ -143,4 +122,32 @@ export default class SharingCredential {
...
 		}
 	}
+	/**
+	 * Fetch a credential from a Verida URI
+	 *
+	 * @param {string} uri
+	 * @return {string} DIDJWT representation of the credential
+	 */
+
+	async fetchCredentialURI(uri: string): Promise<string> {
```
Comment (tahpot) [inline packages/verifiable-credentials/src/sharing-credential.ts:132]: Move this into `client-ts::Utils`

**Review by tahpot — CHANGES_REQUESTED** [review]
@cmcWebCode40 See my review comments.

The unit tests need to also pass.

**Comment by cmcWebCode40** [comment]
@tahpot 
Noted will resolve 

But I tested the unit test and all passed , what error did you get ?

**Comment by tahpot** [comment]
@cmcWebCode40 my mistake. I hadn't rebuilt `client-ts`.

**Comment by tahpot** [comment]
@cmcWebCode40 Once this is done, can you provide instructions on how this works in our documentation?

@pkhien95 Can then use this to generate a URL to display in a QR code on the credential screen in the vault.

**Comment by cmcWebCode40** [comment]
Yes I can provide instructions on how it works , but do i need to edit the docs from notion directly?

**Comment by tahpot** [comment]
Yes, you'll need to edit Notion directly.

Add to the Verifiable Credentials section and then tag me in slack.

**Comment by cmcWebCode40** [comment]
@tahpot, 

i have modified requested changes ,

**Comment by cmcWebCode40** [comment]
@tahpot , 

I don't think i have access to edit our documentation on notion

**Comment by tahpot** [comment]
Some feedback:

- Be consistent with casing `URI` or `Uri`? (I prefer `Uri` as that's what has been used elsewhere)
- `utils.fetchCredentialURI()` -- Need to throw an exception if the item isn't found or if the data can't be decrypted. Add unit tests for these edge cases.
- `utils.getUrlByPath()` -- Rename to `explodeVeridaUri()`
- `utils.generateObjectUri()` -- Rename to `buildVeridaUri()`
- `utils.fetchCredentialURI()` -- Rename to `fetchVeridaUri()`

**Comment by cmcWebCode40** [comment]
noted, will make changes

**Comment by cmcWebCode40** [comment]
@tahpot ,

update requested changes

**Comment by tahpot** [comment]
@cmcWebCode40 This is good work :)

I have made a few final changes, please review and you can then merge.

**Comment by cmcWebCode40** [comment]
Reviewed
